src/flb_unescape.c

```diff
@@ -178,7 +178,7 @@ int flb_unescape_string_utf8(const char *in_buf, int sz, char *out_buf)
                     esc_in = u8_read_escape_sequence(next, size, &ch) + 1;
                 }
                 else {
-                    ch = (uint32_t) *in_buf;
+                    ch = (uint32_t) (signed char) *in_buf;
```

Comment:
Can you add comment about `(signed char)` ?

It is to prevent removing the cast by one who is not familiar with arm like me :).